tests/components/shopping_list/test_init.py

```diff
@@ -19,6 +20,39 @@ async def test_add_item(hass, sl_setup):
     assert response.speech["plain"]["speech"] == "I've added beer to your shopping list"
 
 
+async def test_update_list(hass, sl_setup):
+    """Test updating all list items."""
+    await intent.async_handle(
+        hass, "test", "HassShoppingListAddItem", {"item": {"value": "beer"}}
+    )
+
+    await intent.async_handle(
+        hass, "test", "HassShoppingListAddItem", {"item": {"value": "cheese"}}
+    )
+
+    # Update a single attribute, other attributes shouldn't change
+    await hass.data[DOMAIN].async_update_list({"complete": True})
```

Comment:
It's better to call the services via the core service registry to test this method. Use `hass.services.async_call`.